### ISSUE TITLE
Menu component does not close when pressing escape

### ISSUE THREAD

**Issue by makenosound** [cultureamp/kaizen-design-system#635]
My expectation is that once opened, the menu should close if I press the escape key. This isn’t _quite_ as clear cut a behaviour as it is with modals, but for me at least it feels like that should be the behaviour.

**Comment by dougmacknz** [comment]
I'll tackle this as part of the Select rewrite, which is now looking like a Menu rewrite as well https://github.com/cultureamp/kaizen-design-system/pull/2302
